src/java.base/share/classes/sun/security/util/Pem.java

```diff
@@ -42,13 +120,202 @@ public class Pem {
      * @return the decoded bytes
      * @throws java.io.IOException if input is invalid
      */
-    public static byte[] decode(String input) throws IOException {
+    public static byte[] decode(String input) {
         byte[] src = input.replaceAll("\\s+", "")
-                .getBytes(StandardCharsets.ISO_8859_1);
-        try {
+            .getBytes(StandardCharsets.ISO_8859_1);
             return Base64.getDecoder().decode(src);
-        } catch (IllegalArgumentException e) {
-            throw new IOException(e);
+    }
+
+    /**
+     * Extract the OID from the PBE algorithm.  PBEKS2, which are all AES-based,
+     * has uses one OID for all the standard algorithm, while PBEKS1 uses
+     * individual ones.
+     */
+    public static ObjectIdentifier getPBEID(String algorithm) {
+        try {
+            if (algorithm.contains("AES")) {
```

Comment:
It has been reliable, though I now see that the PBE1 spec does support AES, but SunJCE doesn't.  That's unfortunate as I wouldn't have made that assumption.

Reply:
I deleted my previous comment because it was incorrect, as I mixed up that PBE1 & 2 were in the same spec.
PBES1 is defined with algorithm that are not AES, so any PBE algorithm containing AES is a PBES2 algorithm.  This method is only used when the user gives a PBE algorithm and we must parse it to get the OID.
The alternative to this is having the algorithm names hardcoded in the code doing string compare on each until it finds a match.  I felt this was an overly expensive check for something that should be pretty easy.
I'm not aware of a spec listing other PBES2 algorithms that don't use AES.